data_retriever/migration_event_queue.py

```
@@ -0,0 +1,204 @@
+        try:
+            self._cursor.execute("""
+                INSERT INTO "history_event" (
+                    "entity", "entityId", "action", "metadata", "userAgent", "createdAt"
+                ) VALUES (
+                    'migration', %s, %s, %s, 'UPSTRA', %s
+                );
+                """, (migration_id, serialize_event_type(event), serialize_event(event), datetime.now())
+            )
+            self._conn.commit()
+        except Exception as e:
+            raise EventQueueException(f"Failed to push event to Redis: {e}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)